frontend/src/scenes/dashboard/Dashboard.tsx

```diff
@@ -170,6 +179,14 @@ function DashboardScene(): JSX.Element {
                     <DashboardItems />
                 </div>
             )}
+            {dashboard && (
+                <AddInsightsToDashboardModal
+                    isOpen={addInsightsToDashboardModalOpen}
+                    closeModal={() => setAddInsightsToDashboardModalOpen(false)}
+                    dashboardId={dashboard.id}
+                    canEditDashboard={canEditDashboard}
+                />
```

Comment:
Same here, the props probably don't need to be passed down but can use actions / values from kea. Then you can also get rid of the conditional `dashboard &&`.